src/plugins/intel_cpu/src/nodes/rnn.cpp

```diff
@@ -871,38 +903,49 @@ void RNN::createDescriptor(const std::vector<MemoryDescPtr> &inputDesc,
     supportedPrimitiveDescriptors.emplace_back(config, ref_any);
 }
 
+Node::AttrPtr RNN::initPrimitiveAttr() {
+    auto attr = std::make_shared<dnnl::primitive_attr>(dnnl::primitive_attr());
+    attr->set_scratchpad_mode(dnnl::scratchpad_mode::user);
+
+    if (one_of(getOriginalInputPrecisionAtPort(0), Precision::U8, Precision::I8)) {
+        const int weightsScaleMask = 0;
+
+        attr->set_rnn_weights_qparams(weightsScaleMask, weightsScales);
+        attr->set_rnn_data_qparams(inputScale, inputShift);
+    }
+
+    return attr;
+}
+
 void RNN::prepareParams() {
     for (size_t i = 0; i < wIdx; i++) {
         auto memPtr = getParentEdgesAtPort(i).front()->getMemoryPtr();
         if (!memPtr || !memPtr->isAllocated())
             THROW_ERROR << "has uninitialized memory at port " << i;
     }
 
-    const auto& dataPrecision = getOriginalInputPrecisionAtPort(0);
-    const auto dataType = DnnlExtensionUtils::IEPrecisionToDataType(dataPrecision);
-
     auto dataMemPtr = getParentEdgesAtPort(0).front()->getMemoryPtr();
     const size_t B = dataMemPtr->GetShape().getStaticDims()[0];
     const size_t SL = is_cell ? 1lu : dataMemPtr->GetShape().getStaticDims()[1];
     const Shape shapeS_4D{L, D, B, SC};
 
-    inDataDescs[0] = std::make_shared<DnnlBlockedMemoryDesc>(Shape{SL, B, DC}, dataType, memory::format_tag::tnc);
-    outDataDescs[0] = std::make_shared<DnnlBlockedMemoryDesc>(Shape{SL, B, SC}, dataType, memory::format_tag::tnc);
+    inDataDescs[0] = std::make_shared<DnnlBlockedMemoryDesc>(Shape{SL, B, DC}, inDataTypes[xIdx], memory::format_tag::tnc);
+    outDataDescs[0] = std::make_shared<DnnlBlockedMemoryDesc>(Shape{SL, B, D * SC}, outDataTypes[yIdx], memory::format_tag::tnc);
```

Comment:
Interesting how it worked before. Have we missed some test coverage?

Reply:
We only support D = 1 at the moment, even both ngraph and oneDNN do support bidirectional.
So this change does not really affect anything. Just to highlight which shape is actually required by oneDNN 
To support bidirectional rnn (D = 2) quite a big effort is required from transformations / graph optimizations point of view, because ngraph and oneDNN have different output shape format:
- ngraph: N,D,T,C
- dnnl: T,N,C*D